edgeless_systemtests/src/lib.rs

```diff
@@ -0,0 +1,288 @@
+
+            // The first node in each domain is also assigned a file-log resource.
+            for node_i in 0..num_nodes_per_domain {
+                let (task, handle) = futures::future::abortable(edgeless_node::edgeless_node_main(match node_i {
+                    0 => edgeless_node::EdgelessNodeSettings {
+                        node_id: uuid::Uuid::new_v4(),
+                        agent_url: format!("http://{}:{}", address, next_port()),
+                        invocation_url: format!("http://{}:{}", address, next_port()),
+                        metrics_url: format!("http://{}:{}", address, next_port()),
+                        orchestrator_url: orchestrator_url.to_string(),
+                        http_ingress_url: "".to_string(),
+                        http_ingress_provider: "".to_string(),
+                        http_egress_provider: "".to_string(),
+                        file_log_provider: "file-log-1".to_string(),
+                        redis_provider: "".to_string(),
+                    },
+                    _ => {
+                        edgeless_node::EdgelessNodeSettings::new_without_resources(&orchestrator_url, address, next_port(), next_port(), next_port())
+                    }
+                }));
+                tokio::spawn(task);
+                handles.push(handle);
+            }
+        }
+
+        let (task, handle) = futures::future::abortable(edgeless_con::edgeless_con_main(edgeless_con::EdgelessConSettings {
+            controller_url: controller_url.clone(),
+            orchestrators,
+        }));
+        tokio::spawn(task);
+        handles.push(handle);
+
+        let mut con_client = edgeless_api::grpc_impl::controller::ControllerAPIClient::new(controller_url.as_str()).await;
+
+        tokio::time::sleep(tokio::time::Duration::from_millis(100)).await;
+
+        (handles, con_client.workflow_instance_api())
+    }
+
+    async fn wf_list(client: &mut Box<(dyn WorkflowInstanceAPI)>) -> Vec<edgeless_api::workflow_instance::WorkflowInstance> {
+        match client.list(edgeless_api::workflow_instance::WorkflowId::none()).await {
+            Ok(instances) => instances,
+            Err(_) => vec![],
+        }
+    }
+
+    fn fixture_spec() -> edgeless_api::function_instance::FunctionClassSpecification {
+        edgeless_api::function_instance::FunctionClassSpecification {
+            function_class_id: "system_test".to_string(),
+            function_class_type: "RUST_WASM".to_string(),
+            function_class_version: "0.1".to_string(),
+            function_class_inlude_code: include_bytes!("fixtures/system_test.wasm").to_vec(),
+            outputs: vec!["out1".to_string(), "out2".to_string(), "err".to_string(), "log".to_string()],
+        }
+    }
+
+    fn terminate(handles: Vec<futures::future::AbortHandle>) -> anyhow::Result<()> {
+        for handle in handles {
+            handle.abort();
+        }
+        Ok(())
+    }
+
+    #[tokio::test]
```

Comment:
Given the high number of async tasks, it might actually make sense to test this with more than one thread.

Reply:
Yes, it makes sense. Even better: we could wrap the whole test in a function `real_test()` and then make multiple tests:

```
#[tokio::test(flavor = "multi_thread", worker_threads = 1)]
async fn my_test_1() {
    real_test();
}

#[tokio::test(flavor = "multi_thread", worker_threads = 2)]
async fn my_test_2() {
    real_test();
}

#[tokio::test(flavor = "multi_thread", worker_threads = 10)]
async fn my_test_10() {
    real_test();
}
```

I am not sure if this can help us catch some glitches or would be just just redundant.